developer/src/kmc-ldml/src/compiler/visual-keyboard-compiler.ts

```diff
@@ -3,6 +3,23 @@ import { LDMLKeyboard, CompilerCallbacks } from "@keymanapp/developer-utils";
 import { KeysCompiler } from "./keys.js";
 import { LdmlCompilerMessages } from "./ldml-compiler-messages.js";
 
+// This is a partial polyfill for findLast, so not polluting Array.prototype
+//
+// TODO: remove and replace with Array.prototype.findLast when it is
+// well-supported
+function findLast(arr: any, callback: any) {
+  if (!arr) {
+    return undefined;
+  }
+  const len = arr.length >>> 0;
```

Comment:
Say what? Is this to normalize the value?

Reply:
good question. this is copypasta, so I didn't examine very closely; I found this: https://stackoverflow.com/questions/44946957/what-use-cases-exist-for-using-operator-with-0-unsigned-right-bit-shift